### PR TITLE
[NCLSUP-268] Improve external url validation

### PR DESCRIPTION
**Pull request by dominikbrazdil**
### All Submissions:

* [ ] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/repour/wiki/Changelog) for your change?
* [x] Have you added unit tests for your change?

### CODE REVIEW

**Review by michalovjan — CHANGES_REQUESTED** [review]

File: repour/server/endpoint/external_to_internal.py [hunk 1]
```diff
@@ -14,6 +14,8 @@
     "external_to_internal_req_histogram", "Histogram for external_to_internal endpoint"
 )
 
+SCP_LIKE_URL_REGEX = r"^(\w+://)?(.+):\w(.*)\.git$"  # There is ":" in the middle of url
```
[the remark this answers is not in the document]
Comment (dominikbrazdil) [reply repour/server/endpoint/external_to_internal.py]: Yeah you are right, the .git part in the end should handle this case as the port would probably be behind it but I excluded the numbers to make it more clear

**Review by michalovjan — APPROVED** [review]
LGTM! :)

File: repour/server/endpoint/external_to_internal.py [hunk 2]
```diff
@@ -39,15 +43,11 @@ async def translate_external_to_internal(external_git_url):
     elif not gerrit_server.endswith("/"):
         gerrit_server = gerrit_server + "/"
 
-    is_scp_like = external_git_url.startswith("git@")
+    is_scp_like = re.match(SCP_LIKE_URL_REGEX, external_git_url)
 
-    if is_scp_like:
```
Comment (michalovjan) [inline repour/server/endpoint/external_to_internal.py:44]: wait, why are you deleting this stuff

**Review by michalovjan — CHANGES_REQUESTED** [review]

File: repour/server/endpoint/external_to_internal.py (hunk 2 above)
Comment (dominikbrazdil) [reply repour/server/endpoint/external_to_internal.py:44]: Cause 'git@' is not signature of scp_like url so I cannot split by '@' anymore. So instead the scheme is going to be set manually when the url starts with 'git@'

Comment (dominikbrazdil) [reply repour/server/endpoint/external_to_internal.py:44]: Technically its not scheme but the user..so it's kind of workaround.

**Review by michalovjan — APPROVED** [review]

**Pull request merged**